src/target/breakpoints.c

```diff
@@ -293,36 +293,51 @@ static void breakpoint_free(struct target *data_target, struct target *breakpoin
 	}
 
 	if (!breakpoint)
-		return;
+		return ERROR_BREAKPOINT_NOT_FOUND;
```

Comment:
This is the same case as in watchpoint_free(). The condition of the `if` can't be true, and so the `if` should be replaced by an assert. I will submit this to the upstream shortly. No need to change anything here at the moment.